runtime/vm/createramclass.cpp

```diff
@@ -3861,10 +3882,171 @@ allocateRAMClassFragmentFromFreeList(RAMClassAllocationRequest *request, J9RAMCl
 }
 
 /**
- * Allocates fragments for a RAM class.
- *
- * Returns a pointer to the segment containing the first RAM class fragment allocated.
+ * Allocates fragments from free list.
  */
+static void
+allocateFreeListBlock (RAMClassAllocationRequest *request, J9ClassLoader *classLoader, RAMClassAllocationRequest *prev, J9RAMClassFreeLists *blockFreeLists, UDATA *ramClassUDATABlockFreelist)
```

Comment:
The last two parameters can be combined since `ramClassUDATABlockFreelist ` is now part of the `J9RAMClassFreeLists` structure. Same with the methods `addBlockToFreeList`, `allocateRemainingFragments `, and `allocateRAMClassFragmentFromFreeList `.